src/bootstrap/src/core/download.rs

```diff
@@ -250,19 +257,113 @@ impl Config {
                             "(New-Object System.Net.WebClient).DownloadFile('{}', '{}')",
                             url, tempfile.to_str().expect("invalid UTF-8 not supported with powershell downloads"),
                         ),
-                    ])).is_err() {
+                    ])).is_ok() {
                         return;
                     }
                     eprintln!("\nspurious failure, trying again");
                 }
             }
             if !help_on_error.is_empty() {
                 eprintln!("{help_on_error}");
+                Self::check_outdated(commit);
             }
             crate::exit!(1);
         }
     }
 
+    fn check_outdated(commit: &String) {
+        let check_outdated_msg = || {
+            if !commit.is_empty() {
+                let build_date: String = Command::new("git")
+                    .arg("show")
+                    .arg("-s")
+                    .arg("--format=%ar")
+                    .arg(commit)
+                    .output()
+                    .unwrap()
+                    .stdout
+                    .into_iter()
+                    .map(|c| c as char)
```

Comment:
I think you want `String::from_utf8`, rather than mapping each byte to a char and then converting to a String.